esp/src/eclwatch/PackageMapQueryWidget.js

```diff
@@ -188,39 +188,64 @@ define([
+                                PackageMap: item.Id,
+                                Target: item.Target,
+                                Process: item.Process
+                            }
+                        }).then(function(response){
+                            if (lang.exists("DeletePackageResponse.status", response)) {
+                                if (response.DeletePackageResponse.status.Code === 0) {
+                                    context.refreshGrid();
+                                }
+                            }
+                        });
                     });
                 }
             },
+
             _onActivate: function (event) {
                 var context = this;
-                WsPackageMaps.activatePackageMap(this.packagesGrid.selection.getSelected()).then(function (response) {
-                    context.packagesGrid.rowSelectCell.toggleAllSelection(false);
-                    context.refreshGrid();
-                    return response;
-                }, function (err) {
-                    context.showErrors(err);
-                    return err;
+                var selection = this.packagesGrid.getSelected();
+
+                WsPackageMaps.activatePackageMap({
+                    request: {
+                        Target: selection[0].Target,
+                        Process: selection[0].Process,
+                        PackageMap: selection[0].Id
+                    }
+                }).then(function (response){
+                    if (lang.exists("ActivatePackageResponse.status", response)) {
+                        if (response.ActivatePackageResponse.status.Code === 0) {
+                            context.refreshGrid();
+                        }
+                    }
                 });
             },
             _onDeactivate: function (event) {
                 var context = this;
-                WsPackageMaps.deactivatePackageMap(this.packagesGrid.selection.getSelected()).then(function (response) {
-                    context.packagesGrid.rowSelectCell.toggleAllSelection(false);
-                    context.refreshGrid();
-                    return response;
-                }, function (err) {
-                    context.showErrors(err);
-                    return err;
+                var selection = this.packagesGrid.getSelected();
+
+                WsPackageMaps.deactivatePackageMap({
+                    request: {
+                        Target: selection[0].Target,
+                        Process: selection[0].Process,
+                        PackageMap: selection[0].Id
+                    }
+                }).then(function (response){
+                    if (lang.exists("DeActivatePackageResponse.status", response)) {
+                        if (response.DeActivatePackageResponse.status.Code === 0) {
+                            context.refreshGrid();
+                        }
+                    }
```

Comment:
This looks like it doesn't support the multi-selection?  IOW can you now de-activate several package maps at the same time?